test/jit/test_class_type.py

```diff
@@ -929,3 +929,44 @@ def forward(self, x):
             self.assertEqual(m(input), m_loaded(input))
             # Make sure class constant is accessible from module
             self.assertEqual(m.w, m_loaded.w)
+
+    def test_unused_method(self):
+        """
+        Test unused methods on scripted classes.
+        """
+        @torch.jit.script
+        class Unused(object):
+            def __init__(self):
+                self.count: int = 0
+                self.items: List[int] = []
+
+            def used(self):
+                self.count += 1
+                return self.count
+
+            @torch.jit.unused
+            def unused(self):
+                a = next(self.items)
+                return a
+
+        class ModuleWithUnused(nn.Module):
```

Comment:
Can you add a test to show that the return type / argument types are properly set ?
```
@torch.jit.unused
def unused(self, x: int, y: str) -> int:
  a = next(self.items)
  return a

def hi(self): -> int
   return self.unused(y="hi", x=3)
```